aptos-move/aptos-vm/src/block_executor/vm_wrapper.rs

```diff
@@ -53,29 +53,77 @@ impl<'a, S: 'a + StateView + Sync> ExecutorTask for AptosExecutorTask<'a, S> {
+                // If aggregators are disabled for user transactions, then delta_change_set should be empty as there should not be any delta outputs.
+                assert!(
+                    aggregator_enabled
+                        || !matches!(txn, PreprocessedTransaction::UserTransaction(_))
+                        || vm_output.delta_change_set().is_empty()
+                );
+                if vm_output.status().is_discarded() {
+                    match sender {
+                        Some(s) => speculative_trace!(
+                            &log_context,
+                            format!(
+                                "Transaction discarded, sender: {}, error: {:?}",
+                                s, vm_status
+                            ),
+                        ),
+                        None => {
+                            speculative_trace!(
+                                &log_context,
+                                format!("Transaction malformed, error: {:?}", vm_status),
+                            )
+                        },
+                    };
+                }
+                if AptosVM::should_restart_execution(&vm_output) {
+                    speculative_info!(
+                        &log_context,
+                        "Reconfiguration occurred: restart required".into()
+                    );
+                    ExecutionStatus::SkipRest(AptosTransactionOutput::new(vm_output))
+                } else {
+                    ExecutionStatus::Success(AptosTransactionOutput::new(vm_output))
+                }
+            },
+            Err(err) => ExecutionStatus::Abort(err),
+        }
+    }
+
+    // This function is called by the BlockExecutor for each transaction is intends
+    // to execute (via the ExecutorTask trait). This function is run during
+    // sequential execution of a block.
+    fn execute_transaction_sequential(
         &self,
         view: &impl StateView,
         txn: &PreprocessedTransaction,
         txn_idx: TxnIndex,
-        materialize_deltas: bool,
     ) -> ExecutionStatus<AptosTransactionOutput, VMStatus> {
         let log_context = AdapterLogSchema::new(self.base_view.id(), txn_idx as usize);
 
         match self
             .vm
-            .execute_single_transaction(txn, &view.as_move_resolver(), &log_context)
+            .execute_single_transaction_sequential(txn, &view, &log_context)
         {
-            Ok((vm_status, mut vm_output, sender)) => {
-                if materialize_deltas {
-                    // TODO: Integrate delta application failure.
-                    vm_output = vm_output
-                        .try_materialize(view)
-                        .expect("Delta materialization failed");
-                }
-
+            Ok((vm_status, vm_output, sender)) => {
+                // If aggregators are disabled for user transactions, then delta_change_set should be empty as there should not be any delta outputs.
```

Comment:
Should we have a debug_assert `delta_change_set .is_empty()` or something like that with this comment?

Reply:
Done.